File: Backend/Infrastructure.MyShop/Interfaces/ICategoryService.cs
using Data.MyShop.Entities;
using Infrastructure.MyShop.Models.DTO.CategoryDTO;
using Infrastructure.MyShop.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infrastructure.MyShop.Interfaces
{
    public interface ICategoryService
    {
        Task<ServiceResponse> GetAllAsync();
        Task<ServiceResponse> GetByIdAsync(int id);
        Task<ServiceResponse> GetMainCategoriesAsync();
        //Task<ServiceResponse> GetNearSubcategoriesByCategoryId(int id);
        //Task<List<CategoryItemDTO>> GetAllSubcategoriesByCategoryId(int id);
        Task<ServiceResponse> Create(CategoryCreateDTO model);
        Task<ServiceResponse> EditCategoryAsync(CategoryEditDTO model);
        Task<ServiceResponse> DeleteCategoryAsync(int id);
    }
}

File: Backend/src/main/java/org/example/interfaces/ICategoryService.java
package org.example.interfaces;

import org.example.dto.categoryDTO.CategoryCreateDTO;
import org.example.dto.categoryDTO.CategoryEditDTO;
import org.example.dto.categoryDTO.CategoryItemDTO;

import java.util.List;

public interface ICategoryService {
    CategoryItemDTO create(CategoryCreateDTO model);
    List<CategoryItemDTO> getAll();
    CategoryItemDTO edit(CategoryEditDTO model);
    void deleteById(int id);
    CategoryItemDTO getById(int id);
}
